core/site_battery.go

```diff
@@ -98,6 +98,13 @@ func (site *Site) smartCostActive(lp loadpoint.API, rate api.Rate) bool {
 
 func (site *Site) batteryGridChargeActive(rate api.Rate) bool {
 	limit := site.GetBatteryGridChargeLimit()
+
+	// Grid charging set externally?
+	if site.ValidBatteryModeExternal() {
```

Comment:
This is brittle. The whole external mode reset should be in its own go routine and work independently of any functions being called.